includes/wikia/Wikia.php

```diff
@@ -105,8 +105,12 @@ public static function unsetVar($key) {
 	public static function initAsyncRequest( $wikiId, $user = null ) {
 		$wg = F::app()->wg;
 		$wg->CityID = $wikiId;
-		$wg->DBname = WikiFactory::IDtoDB( $wikiId );
-		$wg->Server = trim( WikiFactory::DBtoUrl( $wg->DBname ), '/' );
+
+		// Do NOT set $wgDbname here.  The wfGetDB method will no longer do
+		// what you think it should since it pulls from a LoadBalance cache
+		// that likely already has cached DB handles for the previous value
+		$dbName = WikiFactory::IDtoDB( $wikiId );
+		$wg->Server = trim( WikiFactory::DBtoUrl( $dbName ), '/' );
```

Comment:
This was the fix.  There was a cached DB handle to the 'wikia' db since this is the db that was set when the script starts up that runs this (e.g. we run with SERVER_ID=177).  When calling:
```php
$wgDBname = 'somewikiname';

...

$db = wfGetDB( DB_SLAVE, null, 'somewikiname' )
```
Code within ```wfGetDB``` notices that the current wiki as given by ```$wgDBname``` is the same as the name passed into the function.  It then clears out the wiki name for the rest of the call, assuming it can always get the name from ```$wgDBname``` if its needs it, and allowing default actions to take place from there on.  This eventually leads it to the LoadBalancer cache where it gets the first connection it finds in the default pool, a pool populated by connections to the 'wikia' DB.